auto_round/__main__.py

```diff
@@ -0,0 +1,420 @@
+                        f"{n} will not be quantized.")
+    lm_head_layer_name = "lm_head"
+    for n, _ in model.named_modules():
+        lm_head_layer_name = n
+    if args.quant_lm_head:
+        from transformers import AutoConfig
+
+        config = AutoConfig.from_pretrained(model_name, trust_remote_code=not args.disable_trust_remote_code)
+        if config.tie_word_embeddings and hasattr(model, "_tied_weights_keys"):
+            tied_keys = model._tied_weights_keys
+            for item in tied_keys:
+                if lm_head_layer_name in item:  ##TODO extend to encoder-decoder layer, seq classification model
+                    args.quant_lm_head = False
+                    print(
+                        f"warning, disable quant_lm_head as quantizing lm_head with tied weights has not been "
+                        f"supported currently")
+                    break
+    if args.quant_lm_head:
+        layer_config[lm_head_layer_name] = {"bits": args.bits}
+        transformers_version = [int(item) for item in transformers.__version__.split('.')[:2]]
+        if transformers_version[0] == 4 and transformers_version[1] < 38:
+            error_message = "Please upgrade transformers>=4.38.0 to support lm-head quantization."
+            raise EnvironmentError(error_message)
+
+    if args.quant_lm_head and args.low_gpu_mem_usage:
+        print(
+            f"warning, low_gpu_mem_usage=False is strongly recommended"
+            " if the whole model could be loaded to gpu")
+
+    autoround = round(
+        model, tokenizer, args.bits, args.group_size, sym=args.sym, batch_size=args.batch_size,
+        dataset=args.dataset, seqlen=seqlen, nblocks=args.nblocks, iters=args.iters, lr=args.lr,
+        minmax_lr=args.minmax_lr, enable_quanted_input=not args.disable_quanted_input, 
+        device=device_str, amp=not args.disable_amp, nsamples=args.nsamples, seed=args.seed,
+        low_gpu_mem_usage=args.low_gpu_mem_usage, scale_dtype=args.scale_dtype, 
+        gradient_accumulate_steps=args.gradient_accumulate_steps, layer_config=layer_config, 
+        enable_minmax_tuning=not args.disable_minmax_tuning, act_bits=args.act_bits,
+        low_cpu_mem_usage=low_cpu_mem_usage, data_type=args.data_type,
+        enable_norm_bias_tuning=args.enable_norm_bias_tuning)
+    model, _ = autoround.quantize()
+    model_name = args.model_name.rstrip("/")
+    if args.low_cpu_mem_mode == 1 or args.low_cpu_mem_mode == 2:
+        import shutil
+
+        shutil.rmtree(args.low_cpu_mem_tmp_dir, ignore_errors=True)
+
+    model.eval()
+    if "cpu" not in device_str:
+        torch.cuda.empty_cache()
+
+    export_dir = args.output_dir + "/" + model_name.split('/')[-1] + f"-autoround-w{args.bits}g{args.group_size}"
+    output_dir = args.output_dir + "/" + model_name.split('/')[-1] + f"-autoround-w{args.bits}g{args.group_size}-qdq"
+
+    eval_folder = f'{export_dir}-{args.format}'
+    if 'round' in args.format or 'gptq' in args.format:
```

Comment:
support exporting to multiple formats, user may need to export to all the formats for a comparison